common/unsafe/src/main/java/org/apache/spark/sql/catalyst/util/CollationFactory.java

```diff
@@ -200,6 +201,7 @@ public Collation(
      * bit 28-24: Reserved.
      * bit 23-22: Reserved for version.
      * bit 21-18: Reserved for space trimming.
+     * 0000 = none, 0001 = left trim, 0010 = right trim, 0011 = trim.
```

Comment:
here's a thought: should we consider having one bit signaling whether there is any trimming sensitivity or not?
while I would agree that it's not immediately useful, it might come in handy in the future

thoughts? cc @nikolamand-db @stefankandic

Reply:
Can you provide some use-case where this one bit would be useful? IMO let's leave one bit reserved, maybe for different class of trimmings (for example not just ascii 20).